packages/react-components/react-tags/src/components/TagGroup/useTagGroup.ts

```diff
@@ -14,10 +16,27 @@ import type { TagGroupProps, TagGroupState } from './TagGroup.types';
 export const useTagGroup_unstable = (props: TagGroupProps, ref: React.Ref<HTMLElement>): TagGroupState => {
   const { onDismiss, size = 'medium' } = props;
 
+  const innerRef = React.useRef<HTMLElement>();
+  const { targetDocument } = useFluent();
+  const { findNextFocusable, findFirstFocusable } = useFocusFinders();
+
   const handleTagDismiss = useEventCallback((e: React.MouseEvent | React.KeyboardEvent, id: string) => {
     onDismiss?.(e, { dismissedTagValue: id });
 
-    // TODO set focus after tag dismiss
+    // set focus after tag dismiss
+    const activeElement = targetDocument?.activeElement;
+    if (innerRef.current?.contains(activeElement as HTMLElement)) {
+      // focus on next tag only if the active element is within the current tag group
+      const next =
+        findNextFocusable(activeElement as HTMLElement, { container: innerRef.current }) ??
+        findFirstFocusable(innerRef.current as HTMLElement);
```

Comment:
personal opinion: how about instead of the first, we do the previous?

It would seem weird to me to dismiss the last option of a multiselect combobox to be back to my first 1 of 10 choices

Reply:
I agree this is more reasonable. I went thru existing cases, when the last one is removed, we either go to the input box or focus just lost. So I think we're free to change this behavior. I'm going to update it to focus on previous instead.